src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java

```diff
@@ -24,21 +26,30 @@
 import org.openstreetmap.atlas.tags.ServiceTag;
 import org.openstreetmap.atlas.tags.TunnelTag;
 import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
 import org.openstreetmap.atlas.utilities.collections.Iterables;
 import org.openstreetmap.atlas.utilities.configuration.Configuration;
 
 /**
  * Flags buildings that intersect/touch centerlines of roads. This doesn't address cases where
- * buildings get really close to roads, but don't overlap them.
+ * buildings get really close to roads, but don't overlap them. The highways that need to be tested
+ * for intersection can be specified in the "highway.filter" config. By default all highways in the
+ * CAR_NAVIGABLE_ENUM_SET of {@link HighwayTag} are checked for intersection.
```

Comment:
Does that mean, if the enum set changes, we'll need to change this checks code as well?

Reply:
I have updated the code to take value from CAR_NAVIGABLE_ENUM_SET, if `highway.filter` is not specified.  The deafult config value is changed to empty string and [this](https://github.com/sayas01/atlas-checks/blob/af2b68685d5eb4579d92dd36023fb5b536928e6a/src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java#L78) has been updated to
```edge -> this.highwayFilter.getValidators().equals(null)? HighwayTag.isCarNavigableHighway(edge): this.highwayFilter.test(edge)```.
Once @alxswny confirms the requirements on configurable, I will update this. Otherwise I will remove the configurable.